### DISCUSSION TITLE
What is connection_builder in ruby kafka?

### DISCUSSION THREAD

**Discussion by SKG0811** [zendesk/ruby-kafka#930] [Q&A] [answered]
This is not a bug. 

I was trying to implement the functionalities of the gem ruby-kafka.

When trying out one of them(fetch_from_partition), I came across the connection_builder. I am finding it hard to understand more about the connection_builder. Any help here is appreciated! Below is the code snippet I was trying. I am sure that I am not doing it right here. Any help is appreciated.

* Version of Ruby: 3.0.0
* Version of Kafka: 2.8.1
* Version of ruby-kafka: 1.4.0

###### Steps to reproduce

```ruby
bp = Kafka::BrokerPool.new(connection_builder: "localhost:9092", logger: Rails.logger)

cluster = Kafka::Cluster.new(seed_brokers: ["localhost:9092"], logger: Rails.logger, resolve_seed_brokers: false, broker_pool: bp)

operation = Kafka::FetchOperation.new(cluster: cluster, logger: Rails.logger, min_bytes: 1,max_wait_time: 10)
operation.fetch_from_partition("test-drive", 6,offset: :latest, max_bytes: 100000)
operation.execute

```

###### Expected outcome

Fetches messages from one or more partitions.

###### Actual outcome

NoMethodError (undefined method `hostname' for "localhost:9092":String)

**Comment by dasch** [comment, marked as answer]
[`ConnectionBuilder`](https://github.com/zendesk/ruby-kafka/blob/master/lib/kafka/connection_builder.rb) is a class that builds connections. You can't just pass in a string.

**Reply by SKG0811** [reply]
Thanks, @dasch for the response.
